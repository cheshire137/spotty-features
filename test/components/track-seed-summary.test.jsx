import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import TrackSearchResponse from '../fixtures/spotify/track-search'

import TrackSeedSummary from '../../src/components/track-seed-summary.jsx'

const track = TrackSearchResponse.tracks.items[0]

function props() {
  return {
    artists: track.artists.map(a => a.name),
    album: track.album.name,
    image: track.album.images[2].url,
    name: track.name
  }
}

describe('TrackSeedSummary', () => {
  let component = null

  beforeEach(() => {
    component = <TrackSeedSummary {...props()} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
