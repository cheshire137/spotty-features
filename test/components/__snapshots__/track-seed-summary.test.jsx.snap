exports[`TrackSeedSummary matches snapshot 1`] = `
<div
  className="columns seed-summary">
  <div
    className="column track-image-column">
    <img
      className="track-image"
      src="https://i.scdn.co/image/b6a166bdf785f16e212ef95984fcf2e788678597" />
  </div>
  <div
    className="column">
    <span
      className="track-name">
      SCREAM
    </span>
    <span>
       by 
    </span>
    <span
      className="track-artists">
      Grimes, Aristophanes
    </span>
    <span
      className="track-album">
      Art Angels
    </span>
  </div>
</div>
`;
